core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -150,11 +165,15 @@ class MetadataCache(brokerId: Int) extends Logging {
   def getTopicMetadata(topics: Set[String],
                        listenerName: ListenerName,
                        errorUnavailableEndpoints: Boolean = false,
-                       errorUnavailableListeners: Boolean = false): Seq[MetadataResponse.TopicMetadata] = {
```

Comment:
Should we remove `MetadataResponse.TopicMetadata` altogether?

Reply:
I think, we can continue to use `MetadataResponse.TopicMetadata` in some of the internal classes and tests. But I open to update/remove the classes the code required.